core/src/telemetry/log_export.rs

```diff
@@ -1,33 +1,35 @@
 use parking_lot::Mutex;
 use ringbuf::{Consumer, HeapRb, Producer};
-use std::{collections::HashMap, sync::Arc, time::SystemTime};
-use temporal_sdk_core_api::telemetry::CoreLog;
+use std::{collections::HashMap, fmt, sync::Arc, time::SystemTime};
+use temporal_sdk_core_api::telemetry::{CoreLog, CoreLogConsumer};
 use tracing_subscriber::Layer;
 
-const RB_SIZE: usize = 2048;
+type CoreLogsOut = Consumer<CoreLog, Arc<HeapRb<CoreLog>>>;
 
-pub(super) type CoreLogsOut = Consumer<CoreLog, Arc<HeapRb<CoreLog>>>;
+#[derive(Debug)]
+struct CoreLogFieldStorage(HashMap<String, serde_json::Value>);
 
-pub(super) struct CoreLogExportLayer {
-    logs_in: Mutex<Producer<CoreLog, Arc<HeapRb<CoreLog>>>>,
+pub(super) struct CoreLogConsumerLayer {
+    consumer: Arc<dyn CoreLogConsumer>,
```

Comment:
Can this just be a `Box`? It's non-pub and doesn't look like it's cloned.

Reply:
I had that originally, but I needed `Arc` from a user for the log pushing since since `Logger` enum is cloneable (because all Telemetry options are). But I guess I can accept `Rc` from a user, `.into_inner()` it, and tell users they are not allowed to hold a reference to it. I will try.

Arguably as options contain more and more real implementations, options should not be cloneable.

Reply:
Sure, fine to have it stay this way then, that's reasonable.